sim/src/recorder.rs

```diff
@@ -29,52 +29,87 @@ impl TrafficRecorder {
 
     pub fn handle_event(&mut self, time: Time, ev: &Event, map: &Map, driving: &DrivingSimState) {
         if let Event::AgentEntersTraversable(AgentID::Car(car), Some(trip), on, _) = ev {
-            if self.seen_trips.contains(trip) {
-                return;
-            }
-            if let Traversable::Lane(l) = on {
-                if self.capture_points.contains(&map.get_l(*l).src_i) {
-                    // Where do they exit?
-                    for step in driving.get_path(*car).unwrap().get_steps() {
-                        if let PathStep::Turn(t) = step {
-                            if self.capture_points.contains(&t.parent) {
-                                self.trips.push(IndividTrip::new(
-                                    time,
-                                    TripPurpose::Shopping,
-                                    TripEndpoint::SuddenlyAppear(Position::start(*l)),
-                                    TripEndpoint::Border(t.parent),
-                                    if car.vehicle_type == VehicleType::Bike {
-                                        TripMode::Bike
-                                    } else {
-                                        TripMode::Drive
-                                    },
-                                ));
-                                self.seen_trips.insert(*trip);
-                                return;
-                            }
-                        }
+            self.on_car_enters_traversable(time, &car, &trip, &on, map, driving);
+        }
+    }
+
+    fn on_car_enters_traversable(
+        &mut self,
+        time: Time,
+        car: &CarID,
```

Comment:
`CarID`, `TripID`, `Traversable` are all `Copy`. I seem to recall clippy / a best practice saying it's roughly the same efficiency to just copy the small struct instead of pass around a reference. I think it'd make the code slightly easier to read; avoiding a bit of dereferencing